app/static/src/app/wodinExcelDownload.ts

```diff
@@ -0,0 +1,99 @@
+import * as XLSX from "xlsx";
+import { Commit } from "vuex";
+import { AppState, AppType } from "./store/appState/state";
+import { FitState } from "./store/fit/state";
+import { AppCtx, Dict } from "./types/utilTypes";
+import { OdinSeriesSet } from "./types/responseTypes";
+import { FitDataGetter } from "./store/fitData/getters";
+import { FitData } from "./store/fitData/state";
+import { ErrorsMutation } from "./store/errors/mutations";
+
+export class WodinExcelDownload {
+    private readonly _state: AppState;
+
+    private readonly _rootGetters: any;
+
+    private readonly _commit: Commit;
+
+    private readonly _fileName: string;
+
+    private readonly _points: number;
+
+    constructor(context: AppCtx, fileName: string, points: number) {
+        this._state = context.rootState;
+        this._rootGetters = context.rootGetters;
+        this._commit = context.commit;
+        this._fileName = fileName;
+        this._points = points;
+    }
+
+    // Shared method to generate both Modelled and Modelled with Data - provide empty nonTimeColumns param to omit data
+    // for Modelled only
+    private static _generateModelledOutput(solutionOutput: OdinSeriesSet, nonTimeColumns: string[],
```

Comment:
You could reduce the complexity of this class by having an overloaded method.

Reply:
Sorry, not sure what you mean - which method do you want to overload?